File: sort/insertion_sort/insertion_sort.py
def insertion_sort(arr, reverse=False):
    for i in range(len(arr)):
        t = arr[i]
        for j in range(i):
            if t < arr[j]:
                for k in range(j, i):
                    arr[k + 1] = arr[k]
                arr[j] = t
                break
    if reverse:
        reverse_arr(arr)


def reverse_arr(arr):
    start, end = 0, len(arr) - 1
    while start < end:
        arr[start], arr[end] = arr[end], arr[start]
        start += 1
        end -= 1

File: sort/insertion_sort/insertion_sort.c
#include <stdio.h>

void insertion_sort(int *arr, int size){
    int t;
    for(int i = 1; i < size; i++){
        if( arr[i - 1] < arr[i] ) continue;
        t = arr[i];
        int j = i - 1;
        for(; j >= 0 && arr[j] > t; j--) arr[j + 1] = arr[j];
        arr[j + 1] = t;
    }
}

void main(){
    int arr[] = {6,5,4,3,2,1,2,3,5,25,32,12,3};
    insertion_sort(arr, 10);
    for(int i = 0; i < 10; i++){
        printf("%d, ", arr[i]);
    }
}